### PR TITLE
Support clipboard write method to support other formats

### PR DESCRIPTION
**Pull request by gupta1**
This PR fixes #
Support clipboard write method so that other formats like html etc are also supported part from plain text

### CODE REVIEW

File: src/vs/platform/clipboard/browser/clipboardService.ts [hunk 1]
```diff
@@ -13,6 +13,10 @@ export class BrowserClipboardService implements IClipboardService {
 
 	private readonly mapTextToType = new Map<string, string>(); // unsupported in web (only in-memory)
 
+	async write(data: any, type?: string): Promise<void> {
+		throw new Error('Not Implemented');
```
Comment (rheh) [inline src/vs/platform/clipboard/browser/clipboardService.ts:17]: Does this need a comment saying why?

File: src/vs/platform/native/common/native.ts [hunk 2]
```diff
@@ -109,6 +109,7 @@ export interface ICommonNativeHostService {
 
 	// Clipboard
 	readClipboardText(type?: 'selection' | 'clipboard'): Promise<string>;
+	writeClipboardData(data: any, type?: 'selection' | 'clipboard'): Promise<void>;
```
Comment (rheh) [inline src/vs/platform/native/common/native.ts:112]: type options are now repeated three times?  Worth add an enum?
`
const enum SelectionType {
	Selection = 'selection',
	Clipboard = 'clipboard'
}
`

Then we can:
`writeClipboardData(data: any, type?: SelectionType): Promise<void>;`

File: src/vs/platform/native/electron-main/nativeHostMainService.ts [hunk 3]
```diff
@@ -473,6 +473,10 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 		return clipboard.readText(type);
 	}
 
+	async writeClipboardData(windowId: number | undefined, data: any, type?: 'selection' | 'clipboard'): Promise<void> {
```
Comment (rheh) [inline src/vs/platform/native/electron-main/nativeHostMainService.ts:476]: Another candidate for using the new suggested enum?

File: src/vs/workbench/services/clipboard/electron-sandbox/clipboardService.ts [hunk 4]
```diff
@@ -20,6 +20,10 @@ export class NativeClipboardService implements IClipboardService {
 		@INativeHostService private readonly nativeHostService: INativeHostService
 	) { }
 
+	async write(data: any, type?: 'selection' | 'clipboard'): Promise<void> {
```
Comment (rheh) [inline src/vs/workbench/services/clipboard/electron-sandbox/clipboardService.ts:23]: Same here for the enum

**Comment by alexdima** [comment]
@gupta1 What is the purpose of this PR? Is the new, added code called from anywhere?

**Review by alexdima — CHANGES_REQUESTED** [review]
I'm not sure what the purpose of this PR is.

**Comment by gupta1** [comment]
> @gupta1 What is the purpose of this PR? Is the new, added code called from anywhere?

@alexdima Yes, it will be called from Azure Data Studio to support copy paste query and results scenario with html formatting.

**Comment by alexdima** [comment]
I have discussed this with the team and they are in agreement that it does not make sense for us to have dead code in vscode. In fact, we have had in the past multiple efforts to remove dead code (e.g. https://github.com/microsoft/vscode/issues/38414). At this time (without a feature making use of this new method), it does not make sense to us to accept the PR. We recommend that you create a PR when the implementations are complete (e.g. the implementation for web currently throws) and when there is a valid use-case for the method in vscode (e.g. a vscode feature that uses the code).

**Pull request closed without merging**